### PR TITLE
ceph-bluestore-tool: create out_dir before create full path  of kvdb

### PR DESCRIPTION
**Pull request by zealoussnow**
Signed-off-by: Leo Zhang <nguzcf@gmail.com>

### CODE REVIEW

**Comment by zealoussnow** [comment]
@liewegas @xiexingguo need a review please

File: src/os/bluestore/bluestore_tool.cc [hunk 1]
```diff
@@ -316,6 +316,13 @@ int main(int argc, char **argv)
       cerr << "readdir in root failed: " << cpp_strerror(r) << std::endl;
       exit(EXIT_FAILURE);
     }
+
+    r = ::mkdir(out_dir.c_str(), 0755);
```
Comment (xiexingguo) [inline src/os/bluestore/bluestore_tool.cc]: We should make sure the dir does not exist before creating? Otherwise this is going to fail..

Comment (zealoussnow) [reply src/os/bluestore/bluestore_tool.cc]: thank for your review, I fixed another mkdir too.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-10-19_12:31:02-rados-wip-kefu-testing-2017-10-19-1823-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-10-19_18:07:02-rados-wip-kefu-testing-2017-10-19-1823-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-10-20_01:29:43-rados-wip-kefu-testing-2017-10-19-1823-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-10-20_03:05:42-rados-wip-kefu-testing-2017-10-19-1823-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-10-20_03:42:46-rados-wip-kefu-testing-2017-10-19-1823-distro-basic-mira/

**Pull request merged**